shpc/main/settings.py

```diff
@@ -114,7 +114,12 @@ def get(self, key, default=None):
         """
         Get a settings value, doing appropriate substitution and expansion.
         """
-        value = self._settings.get(key, default)
+        # This is a reference to a dictionary (object) setting
+        if ":" in key:
+            key, subkey = key.split(":")
+            value = self._settings[key][subkey]
```

Comment:
@vsoch , to be honest, I find this edit of mine to be a bit dirty: the case for a `value` of type dictionary uses a different method than the standard, non-dict case. 
I am not sure why the latter uses the `.get()` method, unlike for instance implementation in the `add` function further down in this source file. 

Ultimately, what I wanted to add here was the ability for this `get` function to handle dict type entries in the settings, same as I am achieving for the `set` function with the other edit in this PR.

Do you think there's a better way to add this functionality?

Reply:
ah, so you gave me the example of what doesn't work, so let me inspect that and either make a suggested change or just for comments! I'm actually working late today and need to have dinner (it's 7pm!) but I'll take a look after dinner!

Reply:
haha okay here is what threw me off (confirming now) the first line seems to reverse the whole thing?
```bash
param[::-1]
'dma:upg:serutaef_reniatnoc'
```
Let me see if I can make something work that is a little more straight forward - I'll update the changelog / version / license headers too. back in a bit!

Reply:
yes, the 1st line reverses the whole string, to split with the right colon, and then the 2nd line individually reverses the two substrings

Reply:
okay take a look! So if you look in the settings.py, the logic was there to parse a nested value, but I had the wrong check - checking the key instead of the value (and indeed the nesting would happen in the value `gpu:amd`. So we didn't need to change config.py under client, just that typo. I also took the opportunity to update the license headers because HAPPY NEW YEAR! :partying_face: